qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -94,19 +136,25 @@ def _compute_gradients(
         """
         res = []
         # compute gradients for all excitation in operator pool
+        sampler = CircuitSampler(vqe.quantum_instance)
```

Comment:
Do you expect that sometimes the same ansatz will be evaluated here? If yes, it could be useful to store one single sampler in the class since it will cache the transpiled circuits

Reply:
The ansatz will change every time so I do not think it makes a difference

Reply:
I ended up extracting the sampler anyways :+1: